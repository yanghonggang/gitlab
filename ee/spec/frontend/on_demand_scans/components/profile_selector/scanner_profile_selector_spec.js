import { mount, shallowMount } from '@vue/test-utils';
import { merge } from 'lodash';
import ProfileSelector from 'ee/on_demand_scans/components/profile_selector/profile_selector.vue';
import OnDemandScansScannerProfileSelector from 'ee/on_demand_scans/components/profile_selector/scanner_profile_selector.vue';
import { scannerProfiles } from '../../mock_data';

const TEST_LIBRARY_PATH = '/test/scanner/profiles/library/path';
const TEST_NEW_PATH = '/test/new/scanner/profile/path';
const TEST_ATTRS = {
  'data-foo': 'bar',
};

describe('OnDemandScansScannerProfileSelector', () => {
  let wrapper;

  const wrapperFactory = (mountFn = shallowMount) => (options = {}) => {
    wrapper = mountFn(
      OnDemandScansScannerProfileSelector,
      merge(
        {
          propsData: {
            profiles: [],
          },
          attrs: TEST_ATTRS,
          provide: {
            scannerProfilesLibraryPath: TEST_LIBRARY_PATH,
            newScannerProfilePath: TEST_NEW_PATH,
          },
        },
        options,
      ),
    );
  };
  const createComponent = wrapperFactory();
  const createFullComponent = wrapperFactory(mount);

  const findProfileSelector = () => wrapper.find(ProfileSelector);

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders properly with profiles', () => {
    createFullComponent({
      propsData: { profiles: scannerProfiles, value: scannerProfiles[0].id },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders properly without profiles', () => {
    createFullComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('sets listeners on profile selector component', () => {
    const inputHandler = jest.fn();
    createComponent({
      listeners: {
        input: inputHandler,
      },
    });
    findProfileSelector().vm.$emit('input');

    expect(inputHandler).toHaveBeenCalled();
  });

  describe('with profiles', () => {
    beforeEach(() => {
      createComponent({
        propsData: { profiles: scannerProfiles },
      });
    });

    it('renders profile selector', () => {
      const sel = findProfileSelector();

      expect(sel.props()).toEqual({
        libraryPath: TEST_LIBRARY_PATH,
        newProfilePath: TEST_NEW_PATH,
        profiles: scannerProfiles.map(x => ({ ...x, dropdownLabel: x.profileName })),
        value: null,
      });
      expect(sel.attributes()).toMatchObject(TEST_ATTRS);
    });
  });
});
