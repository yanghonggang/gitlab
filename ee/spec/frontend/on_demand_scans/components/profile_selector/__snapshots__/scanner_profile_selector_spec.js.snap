// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnDemandScansScannerProfileSelector renders properly with profiles 1`] = `
<div
  class="card"
  data-foo="bar"
  value="gid://gitlab/DastScannerProfile/1"
>
  <!---->
  <div
    class="card-header"
  >
    <div
      class="row"
    >
      <div
        class="col-7"
      >
        <h3
          class="gl-font-lg gl-display-inline"
        >
          Scanner profile
        </h3>
      </div>
       
      <div
        class="col-5 gl-text-right"
      >
        <a
          class="btn btn-success btn-sm gl-button btn-success-secondary"
          data-testid="manage-profiles-link"
          href="/test/scanner/profiles/library/path"
        >
          <!---->
           
          <!---->
            
          <span
            class="gl-button-text"
          >
            
          Manage profiles
        
          </span>
        </a>
      </div>
    </div>
  </div>
  <div
    class="card-body"
  >
    <!---->
    <!---->
      
    <fieldset
      class="form-group gl-form-group"
    >
      <legend
        class="bv-no-focus-ring col-form-label pt-0 col-form-label"
        tabindex="-1"
      >
        Use existing scanner profile
      </legend>
      <div
        class="bv-no-focus-ring"
        role="group"
        tabindex="-1"
      >
         
        <div
          class="dropdown b-dropdown gl-new-dropdown mw-460 btn-group"
          data-testid="profiles-dropdown"
        >
          <!---->
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="btn dropdown-toggle btn-default btn-md gl-button gl-dropdown-toggle"
            type="button"
          >
            <!---->
             
            <!---->
             
            <span
              class="gl-new-dropdown-button-text"
            >
              Scanner profile #1
            </span>
             
            <svg
              aria-hidden="true"
              class="gl-button-icon dropdown-chevron gl-icon s16"
              data-testid="chevron-down-icon"
            >
              <use
                href="#chevron-down"
              />
            </svg>
          </button>
          <ul
            class="dropdown-menu"
            role="menu"
            tabindex="-1"
          >
            <!---->
             
            <li
              class="gl-new-dropdown-item"
              role="presentation"
            >
              <button
                class="dropdown-item"
                role="menuitem"
                type="button"
              >
                <svg
                  class="gl-icon s16 gl-new-dropdown-item-check-icon"
                  data-testid="mobile-issue-close-icon"
                >
                  <use
                    href="#mobile-issue-close"
                  />
                </svg>
                 
                <!---->
                 
                <!---->
                 
                <div
                  class="gl-new-dropdown-item-text-wrapper"
                >
                  <p
                    class="gl-new-dropdown-item-text-primary"
                  >
                    
        Scanner profile #1
      
                  </p>
                   
                  <!---->
                </div>
                 
                <!---->
              </button>
            </li>
            <li
              class="gl-new-dropdown-item"
              role="presentation"
            >
              <button
                class="dropdown-item"
                role="menuitem"
                type="button"
              >
                <svg
                  class="gl-icon s16 gl-new-dropdown-item-check-icon gl-visibility-hidden"
                  data-testid="mobile-issue-close-icon"
                >
                  <use
                    href="#mobile-issue-close"
                  />
                </svg>
                 
                <!---->
                 
                <!---->
                 
                <div
                  class="gl-new-dropdown-item-text-wrapper"
                >
                  <p
                    class="gl-new-dropdown-item-text-primary"
                  >
                    
        Scanner profile #2
      
                  </p>
                   
                  <!---->
                </div>
                 
                <!---->
              </button>
            </li>
             
          </ul>
        </div>
         
        <div
          class="gl-mt-6 gl-pt-6 gl-border-t-solid gl-border-gray-100 gl-border-t-1"
          data-testid="selected-profile-summary"
        >
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="row gl-my-2"
              >
                <div
                  class="col-md-3"
                >
                  Scan mode:
                </div>
                 
                <div
                  class="col-md-9"
                >
                  <strong>
                    Passive
                  </strong>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="row gl-my-2"
              >
                <div
                  class="col-md-3"
                >
                  Spider timeout:
                </div>
                 
                <div
                  class="col-md-9"
                >
                  <strong>
                    5 minutes
                  </strong>
                </div>
              </div>
            </div>
             
            <div
              class="col-md-6"
            >
              <div
                class="row gl-my-2"
              >
                <div
                  class="col-md-3"
                >
                  Target timeout:
                </div>
                 
                <div
                  class="col-md-9"
                >
                  <strong>
                    10 seconds
                  </strong>
                </div>
              </div>
            </div>
          </div>
           
          <div
            class="row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="row gl-my-2"
              >
                <div
                  class="col-md-3"
                >
                  AJAX spider:
                </div>
                 
                <div
                  class="col-md-9"
                >
                  <strong>
                    Off
                  </strong>
                </div>
              </div>
            </div>
             
            <div
              class="col-md-6"
            >
              <div
                class="row gl-my-2"
              >
                <div
                  class="col-md-3"
                >
                  Debug messages:
                </div>
                 
                <div
                  class="col-md-9"
                >
                  <strong>
                    Hide debug messages
                  </strong>
                </div>
              </div>
            </div>
          </div>
        </div>
        <!---->
        <!---->
        <!---->
      </div>
    </fieldset>
     
  </div>
  <!---->
  <!---->
</div>
`;

exports[`OnDemandScansScannerProfileSelector renders properly without profiles 1`] = `
<div
  class="card"
  data-foo="bar"
>
  <!---->
  <div
    class="card-header"
  >
    <div
      class="row"
    >
      <div
        class="col-7"
      >
        <h3
          class="gl-font-lg gl-display-inline"
        >
          Scanner profile
        </h3>
      </div>
       
      <div
        class="col-5 gl-text-right"
      >
        <button
          class="btn btn-success btn-sm disabled gl-button btn-success-secondary"
          data-testid="manage-profiles-link"
          disabled="disabled"
          type="button"
        >
          <!---->
           
          <!---->
            
          <span
            class="gl-button-text"
          >
            
          Manage profiles
        
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="card-body"
  >
    <!---->
    <!---->
      
    <p
      class="gl-text-gray-700"
    >
      No profile yet. In order to create a new scan, you need to have at least one completed scanner profile.
    </p>
     
    <a
      class="btn btn-success btn-md gl-button btn-success-secondary"
      data-testid="create-profile-link"
      href="/test/new/scanner/profile/path"
    >
      <!---->
       
      <!---->
        
      <span
        class="gl-button-text"
      >
        Create a new scanner profile
      </span>
    </a>
     
  </div>
  <!---->
  <!---->
</div>
`;
